src/main/java/com/google/firebase/cloud/StorageClient.java

```diff
@@ -0,0 +1,144 @@
+   * configured via {@link com.google.firebase.FirebaseOptions} when initializing the app. If
+   * no bucket was configured via options, this method throws an exception.
+   *
+   * @return a cloud storage Bucket instance, or null if the configured bucket does not exist.
+   * @throws IllegalArgumentException If no bucket is configured via <code>FirebaseOptions</code>,
+   *     or if the bucket does not exist.
+   */
+  public Bucket getBucket() {
+    return getBucket(app.getOptions().getStorageBucket());
+  }
+
+  /**
+   * Returns a cloud storage Bucket instance for the specified bucket name.
+   *
+   * @param name a non-null, non-empty bucket name.
+   * @return a cloud storage Bucket instance, or null if the specified bucket does not exist.
+   * @throws IllegalArgumentException If the bucket name is null, empty, or if the specified
+   *     bucket does not exist.
+   */
+  public Bucket getBucket(String name) {
+    checkArgument(!Strings.isNullOrEmpty(name),
+        "Bucket name not specified. Specify the bucket name via the storageBucket "
+            + "option when initializing the app, or specify the bucket name explicitly when "
+            + "calling the getBucket() method.");
+    Bucket bucket = storage.get(name);
+    checkArgument(bucket != null, "Bucket " + name + " does not exist.");
+    return bucket;
+  }
+
+  /**
+   * Returns a long-lived download URL with a revokable token. This can be used to share a file
+   * with others, but can be revoked by a developer in the Firebase console if desired.
+   *
+   * @param blob The blob to generate a download URL for.
+   * @return a URL string.
+   */
+  public String getDownloadUrl(Blob blob) {
+    checkNotNull(blob, "Blob must not be null");
+    Map<String, String> metadata = blob.getMetadata();
+    if (metadata == null) {
+      metadata = new HashMap<>();
+    }
+    String downloadTokens = metadata.get(DOWNLOAD_TOKENS);
+    if (Strings.isNullOrEmpty(downloadTokens)) {
+      downloadTokens = UUID.randomUUID().toString();
+      metadata.put(DOWNLOAD_TOKENS, downloadTokens);
+      blob.toBuilder().setMetadata(metadata).build().update();
+    }
+
+    String token = downloadTokens.split(",")[0];
+    return String.format("https://firebasestorage.googleapis.com/v0/b/%s/o/%s?alt=media&token=%s",
+        blob.getBucket(), blob.getName(), token);
+  }
+
+  private static final String SERVICE_ID = StorageClient.class.getName();
+
+  private static class StorageClientService extends FirebaseService<StorageClient> {
+
+    StorageClientService(FirebaseApp app) {
+      super(SERVICE_ID, new StorageClient(app));
+    }
+
+    @Override
+    public void destroy() {
+      // NOTE: We don't explicitly tear down anything here, but public methods of StorageClient
+      // will now fail because calls to getOptions() and getToken() will hit FirebaseApp,
+      // which will throw once the app is deleted.
```

Comment:
`getBucket()` calls `app.getOptions()`. Also `StorageClient` is authenticated with a `FirebaseOAuthCredentials`, which calls `app.getToken()` internally. These `app` invocations fail when the app is deleted.